hs_core/views/resource_folder_hierarchy.py

```diff
@@ -1,456 +1,697 @@
+        return HttpResponse(ex.stderr, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except DRF_ValidationError as ex:
+        return HttpResponse(ex.detail, status=status.HTTP_400_BAD_REQUEST)
+
+    return_object = {'new_folder_rel_path': folder_path}
+
+    return HttpResponse(
+        json.dumps(return_object),
+        content_type="application/json"
+    )
+
+
+def data_store_remove_folder(request):
+    """
+    remove a sub-folder/sub-collection in hydroshareZone or any federated zone used for HydroShare
+    resource backend store. It is invoked by an AJAX call and returns json object that include a
+    status of 'success' if succeeds, and HttpResponse of status code of 403, 400, or 500 if fails.
+    The AJAX request must be a POST request with input data passed in for res_id and folder_path
+    where folder_path is the relative path for the folder to be removed under
+    res_id collection/directory.
+    """
+    res_id = request.POST.get('res_id', None)
+    if res_id is None:
+        return HttpResponse('Bad request - resource id is not included',
+                            status=status.HTTP_400_BAD_REQUEST)
+    res_id = str(res_id).strip()
+    try:
+        resource, _, user = authorize(request, res_id,
+                                      needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+    except NotFound:
+        return HttpResponse('Bad request - resource not found', status=status.HTTP_400_BAD_REQUEST)
+    except PermissionDenied:
+        return HttpResponse('Permission denied', status=status.HTTP_401_UNAUTHORIZED)
+
+    folder_path = request.POST.get('folder_path', None)
+    if folder_path is None:
+        return HttpResponse('Bad request - folder_path is not included',
+                            status=status.HTTP_400_BAD_REQUEST)
+    folder_path = str(folder_path).strip()
+    if not folder_path:
+        return HttpResponse('Bad request - folder_path cannot be empty',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if not folder_path.startswith('data/contents/'):
+        return HttpResponse('Bad request - folder_path must start with data/contents/',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    if folder_path.find('/../') >= 0 or folder_path.endswith('/..'):
+        return HttpResponse('Bad request - folder_path must not contain /../',
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        remove_folder(user, res_id, folder_path)
+    except SessionException as ex:
+        return HttpResponse(ex.stderr, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+    except Exception as ex:
+        return HttpResponse(ex.message, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    return_object = {'status': 'success'}
+    return HttpResponse(
+        json.dumps(return_object),
+        content_type="application/json"
+    )
+
+
+def data_store_file_or_folder_move_or_rename(request, res_id=None):
```

Comment:
I did not change anything. I instead replaced it in the internal API. The internal endpoint is no longer called from the drag and drop ui.